apps/site/assets/ts/schedule/components/direction/ScheduleDirectionMenu.tsx

```diff
@@ -53,10 +30,35 @@ const ScheduleDirectionMenu = ({
 }: Props): ReactElement<HTMLElement> => {
   const routePatterns = routePatternsByDirection[directionId];
 
-  useEffect(() => {
-    externalCloseEvent(dispatch);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  useEffect(
+    () => {
+      if (!menuOpen) return () => {};
+      // if the menu is open, add event listeners to enable closing the menu by
+      // clicking outside the menu or with keypress
+      const clickToClose = (event: Event): void => {
+        if (!event.target) return;
+        const element = event.target as HTMLElement;
+        if (element.closest(".js-m-schedule-click-boundary")) return;
+        dispatch(closeRoutePatternMenuAction());
```

Comment:
Potential future refactor: we could pass a `closeMenu` function into `ScheduleDirectionMenu`. That way it wouldn't have to know anything about `dispatch` or `closeRoutePatternMenuAction` which aren't really its concern.